view/adminhtml/templates/landingpage/search-configuration.phtml

```diff
@@ -301,490 +297,501 @@ $isConfig = [
+            });
         }
 
-        $('input[name="configuration"]').val(JSON.stringify(formattedConfig)).change();
-        $('input[name="algolia_configuration"]').val(JSON.stringify(formattedConfig)).change();
-        $('input[name="price_key"]').val(getCurrentPriceKey()).change();
-      }
-
-      function getCurrentIndexName() {
-        return config.indexDataByStoreIds[$('select[name="store_id"]').val()].indexName + '_products';
-      }
-
-      function getCurrentPriceKey() {
-        return config.indexDataByStoreIds[$('select[name="store_id"]').val()].priceKey;
-      }
-
-      var customAttributeFacet = {
-        categories: function (facet, templates) {
-          var hierarchical_levels = [];
-          for (var l = 0; l < 10; l++)
-            hierarchical_levels.push('categories.level' + l.toString());
-
-          var hierarchicalMenuParams = {
-            container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-            attributes: hierarchical_levels,
-            separator: config.categorySeparator,
-            limit: config.maxValuesPerFacet,
-            templates: templates,
-            sortBy: ['name:asc'],
-            cssClasses: {
-              list: 'hierarchical',
-              root: 'facet hierarchical'
-            }
-          };
+        initInstantSearch();
 
-          hierarchicalMenuParams.templates.item = '' +
-            '<div class="ais-hierearchical-link-wrapper">' +
-            '<a class="{{cssClasses.link}}" href="{{url}}">{{label}}' +
-            '{{#isRefined}}<span class="cross-circle"></span>{{/isRefined}}' +
-            '<span class="{{cssClasses.count}}">{{count}}</span></a>' +
-            '</div>';
-
-          return ['hierarchicalMenu', hierarchicalMenuParams];
-        }
-      };
-
-      window.getFacetWidget = function (facet, templates) {
-
-        if (facet.type === 'priceRanges') {
-          delete templates.item;
-
-          return ['priceRanges', {
-            container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-            attributeName: facet.attribute,
-            labels: {
-              currency: config.currencyCode,
-              separator: 'to',
-              button: 'go'
-            },
-            templates: templates,
-            cssClasses: {
-              root: 'facet conjunctive'
-            }
-          }];
-        }
-
-        if (facet.type === 'conjunctive') {
-          var refinementListOptions = {
-            container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-            attributeName: facet.attribute,
-            limit: config.maxValuesPerFacet,
-            operator: 'and',
-            templates: templates,
-            sortBy: ['count:desc', 'name:asc'],
-            cssClasses: {
-              root: 'facet conjunctive'
-            }
-          };
+        $('select[name="store_id"]').on('change', function(e) {
+            initInstantSearch();
```

Comment:
Does `initAutocomplete` need to be called as well? How do we ensure that it is pulling from the correct application?

Reply:
`initAutocomplete` is defined and called in `initInstantSearch` :
```
L624-629: 
search.on('render', function() {
      initAutocomplete();
      initSortableTable();
});

search.start();
```
This is how it was designed originally (calling `initAutocomplete` in the `search.on()` method), I'm not sure if it would be considered as best practice on a frontend development point of view though but ... it works ;) .

Reply:
Ah... I missed that call there. Works for me. Thanks for explaining!